apps/language_server/test/experimental/provider/handlers/goto_definition_test.exs

```diff
@@ -0,0 +1,215 @@
+defmodule ElixirLS.Experimental.Provider.Handlers.GotoDefinitionTest do
+  use ExUnit.Case, async: true
+
+  alias ElixirLS.LanguageServer.Experimental.Protocol.Requests.GotoDefinition
+  alias ElixirLS.LanguageServer.Experimental.Protocol.Responses
+  alias ElixirLS.LanguageServer.Experimental.Provider.Env
+  alias ElixirLS.LanguageServer.Experimental.Provider.Handlers
+  alias ElixirLS.LanguageServer.Experimental.SourceFile
+  alias ElixirLS.LanguageServer.Experimental.SourceFile.Conversions
+
+  alias ElixirLS.LanguageServer.Fixtures.LspProtocol
+  alias ElixirLS.LanguageServer.Test.FixtureHelpers
+
+  import LspProtocol
+  import ElixirLS.Test.TextLoc, only: [annotate_assert: 4]
+
+  setup do
+    {:ok, _} = start_supervised(SourceFile.Store)
+    :ok
+  end
+
+  def request(file_path, line, char) do
+    uri = Conversions.ensure_uri(file_path)
+
+    params = [
+      text_document: [uri: uri],
+      position: [line: line, character: char]
+    ]
+
+    with {:ok, contents} <- File.read(file_path),
+         :ok <- SourceFile.Store.open(uri, contents, 1),
+         {:ok, _source_file} <- SourceFile.Store.fetch(uri),
+         {:ok, req} <- build(GotoDefinition, params) do
+      GotoDefinition.to_elixir(req)
+    end
+  end
+
+  def handle(request) do
+    Handlers.GotoDefinition.handle(request, Env.new())
+  end
+
+  defp arrange_referenced_file do
+    file_path = FixtureHelpers.get_path("references_referenced.ex")
+    uri = Conversions.ensure_uri(file_path)
+    %{uri: uri}
+  end
+
+  test "find definition remote function call" do
+    b_file = arrange_referenced_file()
+    file_path = FixtureHelpers.get_path("references_remote.ex")
+    {line, char} = {4, 28}
+
+    {:ok, request} = request(file_path, line, char)
+
+    annotate_assert(file_path, line, char, """
+        ReferencesReferenced.referenced_fun()
+                                ^
+    """)
+
+    {:reply, %Responses.GotoDefinition{result: definition}} = handle(request)
+
+    assert definition.uri == b_file.uri
+    assert definition.range.start.line == 1
+    assert definition.range.start.character == 6
+    assert definition.range.end.line == 1
+    assert definition.range.end.character == 6
+  end
+
+  test "find definition remote macro call" do
+    b_file = arrange_referenced_file()
+    file_path = FixtureHelpers.get_path("references_remote.ex")
+    {line, char} = {8, 28}
+
+    {:ok, request} = request(file_path, line, char)
+
+    annotate_assert(file_path, line, char, """
+        ReferencesReferenced.referenced_macro a do
+                                ^
+    """)
+
+    {:reply, %Responses.GotoDefinition{result: definition}} = handle(request)
```

Comment:
might want to put an assert before this match.

Reply:
I don't think we should use 
`assert {:reply, %Responses.GotoDefinition{result: definition}} = handle(request)` 
instead of `{:reply, %Responses.GotoDefinition{result: definition}} = handle(request)`, 

becuase this is the real `act` of this testing, above this line is arrange block, and below it is assertion. Not using assert in this line provides a better splitting and reading experience.

I'd like to hear your reasons for wanting to do this, and in this [file](https://github.com/elixir-lsp/elixir-ls/blob/master/apps/language_server/test/experimental/provider/handlers/find_references_test.exs#L68), you didn't do so.

Reply:
because if it fails, you'll get a nice diff and not a generic `match_error`

Reply:
alright, you convinced me. I was worried about the match error that would be generated, but it's pretty clear.